# Solar Insights AI: Data Understanding

This Jupyter notebook aims to explore and analyze solar energy data from the Togo-Dapaong_qc dataset. The objective is to gain insights into the dataset's features and basic information. 

Key steps in this analysis include:
1. Loading and inspecting the dataset
2. Examining basic statistics and data distributions
3. Checking for missing values


In [1]:
# Import necessary libraries
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
# Load the Togo dataset
df_togo = pd.read_csv('data/togo-dapaong_qc.csv')

In [3]:
# Display the first five rows
print("First five rows of the dataset:")
display(df_togo.head())

First five rows of the dataset:


Timestamp  GHI  DNI  DHI  ModA  ModB  Tamb    RH   WS  WSgust  \
0  2021-10-25 00:01 -1.3  0.0  0.0   0.0   0.0  24.8  94.5  0.9     1.1   
1  2021-10-25 00:02 -1.3  0.0  0.0   0.0   0.0  24.8  94.4  1.1     1.6   
2  2021-10-25 00:03 -1.3  0.0  0.0   0.0   0.0  24.8  94.4  1.2     1.4   
3  2021-10-25 00:04 -1.2  0.0  0.0   0.0   0.0  24.8  94.3  1.2     1.6   
4  2021-10-25 00:05 -1.2  0.0  0.0   0.0   0.0  24.8  94.0  1.3     1.6   

   WSstdev     WD  WDstdev   BP  Cleaning  Precipitation  TModA  TModB  \
0      0.4  227.6      1.1  977         0            0.0   24.7   24.4   
1      0.4  229.3      0.7  977         0            0.0   24.7   24.4   
2      0.3  228.5      2.9  977         0            0.0   24.7   24.4   
3      0.3  229.1      4.6  977         0            0.0   24.7   24.4   
4      0.4  227.5      1.6  977         0            0.0   24.7   24.4   

   Comments  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [4]:
#remove comment column 
del df_togo['Comments']

# Get a summary of the data
print("\nDataFrame Info:")
df_togo.info()


DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 525600 entries, 0 to 525599
Data columns (total 18 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   Timestamp      525600 non-null  object 
 1   GHI            525600 non-null  float64
 2   DNI            525600 non-null  float64
 3   DHI            525600 non-null  float64
 4   ModA           525600 non-null  float64
 5   ModB           525600 non-null  float64
 6   Tamb           525600 non-null  float64
 7   RH             525600 non-null  float64
 8   WS             525600 non-null  float64
 9   WSgust         525600 non-null  float64
 10  WSstdev        525600 non-null  float64
 11  WD             525600 non-null  float64
 12  WDstdev        525600 non-null  float64
 13  BP             525600 non-null  int64  
 14  Cleaning       525600 non-null  int64  
 15  Precipitation  525600 non-null  float64
 16  TModA          525600 non-null  float64
 17  TModB       

In [5]:
# Get summary statistics
print("\nSummary Statistics:")
display(df_togo.describe())


Summary Statistics:


GHI            DNI            DHI           ModA  \
count  525600.000000  525600.000000  525600.000000  525600.000000   
mean      230.555040     151.258469     116.444352     226.144375   
std       322.532347     250.956962     156.520714     317.346938   
min       -12.700000       0.000000       0.000000       0.000000   
25%        -2.200000       0.000000       0.000000       0.000000   
50%         2.100000       0.000000       2.500000       4.400000   
75%       442.400000     246.400000     215.700000     422.525000   
max      1424.000000    1004.500000     805.700000    1380.000000   

                ModB           Tamb             RH             WS  \
count  525600.000000  525600.000000  525600.000000  525600.000000   
mean      219.568588      27.751788      55.013160       2.368093   
std       307.932510       4.758023      28.778732       1.462668   
min         0.000000      14.900000       3.300000       0.000000   
25%         0.000000      24.200000      26.500000       1.400000   
50%         4.300000      27.200000      59.300000       2.200000   
75%       411.000000      31.100000      80.800000       3.200000   
max      1367.000000      41.400000      99.800000      16.100000   

              WSgust        WSstdev             WD        WDstdev  \
count  525600.000000  525600.000000  525600.000000  525600.000000   
mean        3.229490       0.557740     161.741845      10.559568   
std         1.882565       0.268923      91.877217       5.915490   
min         0.000000       0.000000       0.000000       0.000000   
25%         1.900000       0.400000      74.800000       6.900000   
50%         2.900000       0.500000     199.100000      10.800000   
75%         4.400000       0.700000     233.500000      14.100000   
max        23.100000       4.700000     360.000000      86.900000   

                  BP       Cleaning  Precipitation          TModA  \
count  525600.000000  525600.000000  525600.000000  525600.000000   
mean      975.915242       0.000535       0.001382      32.444403   
std         2.153977       0.023116       0.026350      10.998334   
min       968.000000       0.000000       0.000000      13.100000   
25%       975.000000       0.000000       0.000000      23.900000   
50%       976.000000       0.000000       0.000000      28.400000   
75%       977.000000       0.000000       0.000000      40.600000   
max       983.000000       1.000000       2.300000      70.400000   

               TModB  
count  525600.000000  
mean       33.543330  
std        12.769277  
min        13.100000  
25%        23.600000  
50%        28.400000  
75%        43.000000  
max        94.600000

In [6]:
# Check for missing values
print("\nMissing Values in Each Column:")
missing_values = df_togo.isnull().sum()
display(missing_values)


Missing Values in Each Column:


Timestamp        0
GHI              0
DNI              0
DHI              0
ModA             0
ModB             0
Tamb             0
RH               0
WS               0
WSgust           0
WSstdev          0
WD               0
WDstdev          0
BP               0
Cleaning         0
Precipitation    0
TModA            0
TModB            0
dtype: int64

In [ ]:
# Visualize missing values
plt.figure(figsize=(12, 6))
sns.heatmap(df_togo.isnull(), cbar=False, cmap='viridis')
plt.title('Missing Values Heatmap')
plt.show()

### Key Findings

1. **Data Completeness**:
   - The dataset is complete with no missing values in any of the columns.

2. **Data Distribution**:
   - The summary statistics provide insights into the distribution of the data. For example:
     - `GHI` (Global Horizontal Irradiance) has a mean of approximately 230.56 and a standard deviation of 322.53.
     - `DNI` (Direct Normal Irradiance) has a mean of approximately 151.26 and a standard deviation of 250.96.
     - `Tamb` (Ambient Temperature) has a mean of approximately 27.75°C and a standard deviation of 4.76°C.
     - `RH` (Relative Humidity) has a mean of approximately 55.01% and a standard deviation of 28.78%.

3. **Data Range**:
   - The data ranges for various columns are provided, such as:
     - `GHI` ranges from -12.7 to 1424.0.
     - `DNI` ranges from 0.0 to 1004.5.
     - `Tamb` ranges from 14.9°C to 41.4°C.
     - `RH` ranges from 3.3% to 99.8%.